### PR TITLE
fix: change randomizing function

### PR DESCRIPTION
**Pull request by egentilucci**
Hi @DonNicoJs, I found out that the randomizing function, based on the length of the "workingDeck" array, wasn't guaranteeing for the same number to happen only once (i.e. the probability of number 0 to come out was always present until the array has at least 1 element).

To overcome this I added a function stat shuffle the starting deck making it the "workingDeck". The "workingDeck" is then iterated through and after the first card is shown and the user moves to the next, the card is removed from the array (every time only the card at index 0 is removed.

In addition a took the liberty of adding a <div> showing the remaining cards and after the last card is revealed the app shows a card which contains the text "FINISHED".

Let me know what do you think about it.

Thank you in advance for taking the time to review this PR.

Have a nice day.

### CODE REVIEW

File: src/components/QuestionDeck.vue [hunk 1]
```diff
@@ -1,5 +1,11 @@
 <template>
-  <q-card class="my-card bg-grey-9 text-white" bordered>
```
Comment (nmezzopera) [inline src/components/QuestionDeck.vue:2]: @egentilucci This change seems unrelated to the randomizing function

File: src/components/QuestionDeck.vue [hunk 2]
```diff
@@ -20,34 +26,51 @@
       </q-btn>
     </q-card-actions>
   </q-card>
+  <q-card class="my-card bg-grey-9 text-white" align="center" bordered v-else>
+    FINISHED
+  </q-card>
 </template>
 
 <script lang="ts" setup>
-import { defineProps, ref, computed } from 'vue';
```
Comment (nmezzopera) [inline src/components/QuestionDeck.vue:26]: @egentilucci why are we removing defineProps when we are using it on line 54?

File: src/components/QuestionDeck.vue (hunk 1 above)
Comment (egentilucci) [reply src/components/QuestionDeck.vue:2]: That's correct, it is not related to the randomizing function. It is part of the change I made to show a card with the text "FINISHED" when the user passed through all the cards in the deck. I made it contextually to the fix to the randomizing function because it was useful for testing it.
Maybe it would be cleaner to have it in a specific branch though. I can make the change if needed.

File: src/components/QuestionDeck.vue (hunk 2 above)
Comment (egentilucci) [reply src/components/QuestionDeck.vue:26]: I was receiving a warning about using it both as run-time declaration and type declaration that's noted [here](https://vuejs.org/api/sfc-script-setup.html#type-only-props-emit-declarations) in the official documentation where it says that:

> defineProps or defineEmits can only use either runtime declaration OR type declaration. Using both at the same time will result in a compile error.

I took the chance of this PR to remove the double declaration.
Same as above, if it's usually better to move this kind of change to a specific PR I'll be happy to move it in order to make everything as clean as possible.

File: src/components/QuestionDeck.vue (hunk 1 above)
Comment (nmezzopera) [reply src/components/QuestionDeck.vue:2]: Yes please let's separate this into another PR

File: src/components/QuestionDeck.vue (hunk 2 above)
Comment (nmezzopera) [reply src/components/QuestionDeck.vue:26]: Nice catch, worth exploring that warning and why we are loading  stuff globably
